aptos-move/framework/aptos-framework/sources/governed_gas_pool.move

```diff
@@ -20,15 +21,28 @@ module aptos_framework::governed_gas_pool {
     #[test_only]
     use aptos_framework::aptos_coin::Self;
 
+    friend aptos_framework::stake;
+
     const MODULE_SALT: vector<u8> = b"aptos_framework::governed_gas_pool";
 
+    /// Event emitted when token are withdraw from the pool
+    struct WithdrawStakingRewardEvent has drop, store {
+        amount: u64,
+    }
+
     /// The Governed Gas Pool
     /// Internally, this is a simply wrapper around a resource account. 
     struct GovernedGasPool has key {
         /// The signer capability of the resource account.
         signer_capability: SignerCapability,
     }
 
+    /// Contains added variable needed for the GovernedGasPool staking reward update.
+    struct GovernedGasPoolV2 has key {
```

Comment:
naming is hard. But this is not ggp v2. this is more like a GovernedGasPoolExtension.